### PR TITLE
Add more logs in the archive settings.

### PR DESCRIPTION
**Pull request by crummel**
I noticed some missing logs while doing the master branch work - not all repos are using binlogs yet so we're missing out on some logs.  This also adds more logs that source-build itself produces.

### CODE REVIEW

**Comment by crummel** [comment]
Yes, I think that should be an easy enough change, I'll try it out.
@dotnet-bot test CI please

**Comment by crummel** [comment]
@dotnet-bot test CI please
@dotnet-bot test OSX10.12 Release please

**Comment by crummel** [comment]
@dotnet-bot test CI please

**Comment by crummel** [comment]
@dotnet-bot test CI please

**Comment by crummel** [comment]
@dotnet-bot test CI please

**Comment by crummel** [comment]
@dotnet-bot test CI please

File: netci.groovy [hunk 1]
```diff
@@ -3,14 +3,36 @@ import jobs.generation.ArchivalSettings;
...
+  archivalSettings.addFiles("source-build/init-tools.log")
+  archivalSettings.addFiles("source-build/msbuild.log")
+  archivalSettings.addFiles("source-build/testing-smoke/smoke-test.log")
+  archivalSettings.addFiles("tarball-output/bin/logs/*")
+  archivalSettings.addFiles("tarball-output/bin/logs/**/*")
+  archivalSettings.addFiles("tarball-output/src/**/*.binlog")
+  archivalSettings.addFiles("tarball-output/src/**/*.log")
+  archivalSettings.addFiles("tarball-output/init-tools.log")
+  archivalSettings.addFiles("tarball-output/msbuild.log")
+  archivalSettings.addFiles("tarball-output/testing-smoke/smoke-test.log")
```
Comment (dagood) [inline netci.groovy]: Maybe you're just in the middle of trying things out, but I suggest deduping with something like:

```groovy
["source-build", "tarball-output"].each { logRoot ->
```

Comment (crummel) [reply netci.groovy]: Thanks, I will definitely clean this up before checking in.

**Comment by crummel** [comment]
@dotnet-bot test CI please

**Comment by crummel** [comment]
@dotnet-bot test CI please

**Comment by crummel** [comment]
@dotnet-bot test RHEL7.2 Tarball Release please

**Pull request merged**